controllers/horizon_controller.go

```diff
@@ -586,10 +586,10 @@ func (r *HorizonReconciler) ensureHorizonSecret(
 	//
 	// check if secret already exist
 	//
-	_, _, err := oko_secret.GetSecret(ctx, h, horizon.ServiceName, instance.Namespace)
+	scrt, _, err := oko_secret.GetSecret(ctx, h, horizon.ServiceName, instance.Namespace)
 	if err != nil && !k8s_errors.IsNotFound(err) {
 		return err
-	} else if k8s_errors.IsNotFound(err) {
+	} else if k8s_errors.IsNotFound(err) || len(scrt.Data["horizon-secret"]) == 0 {
```

Comment:
Should we fail in case secret is empty ? I expect some users might want to inject their own secret manually then we should warn the user that the given secret is incorrect instead of overriding it silently.

Reply:
Also, it might be helpful for future maintenance if we can create a separate validateHorizonSecret function to gather all logics we need for validation.

Reply:
So, I updated just the function part at the moment. Moved the check into a function like you suggested.

To ensure the conversation is also captured here for posterity. We will implement a mechanism that will allow a user to manually define this secret via the Spec. In which case, we will not override user provided secrets. Such a check will need to validate that:

1. The secret exists;
2. The secret IS owned by the operator

Else, we will fail and update the `Status.Condition` to make users aware of invalid Horizon secrets, such as empty ones in this case.